### ISSUE TITLE
my config for HP Pavilion dv7

### ISSUE THREAD

**Issue by nickbrick** [hirschmann/nbfc#137]
Hello. This config will control the fan on a dv7 by changing the register that contains the thermal zone temperature, thereby fooling the fan into spinning faster or slower. I haven't been able to locate the register that contains the actual fan speed yet. Also, the operation isn't very smooth as the temperature sensor has time to refresh, replacing the fake value with the actual one, so the fan speed will jump between user-desired and system-intended.
```
<?xml version="1.0"?>
<FanControlConfigV2 xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <NotebookModel>HP Pavilion dv7 Notebook PC</NotebookModel>
  <Author>nickbrick</Author>
  <EcPollInterval>100</EcPollInterval>
  <ReadWriteWords>false</ReadWriteWords>
  <CriticalTemperature>75</CriticalTemperature>
  <FanConfigurations>
    <FanConfiguration>
      <ReadRegister>176</ReadRegister>
      <WriteRegister>176</WriteRegister>
      <MinSpeedValue>0</MinSpeedValue>
      <MaxSpeedValue>5</MaxSpeedValue>
      <IndependentReadMinMaxValues>false</IndependentReadMinMaxValues>
      <MinSpeedValueRead>0</MinSpeedValueRead>
      <MaxSpeedValueRead>100</MaxSpeedValueRead>
      <ResetRequired>false</ResetRequired>
      <FanSpeedResetValue>0</FanSpeedResetValue>
      <FanDisplayName>ctmp</FanDisplayName>
      <TemperatureThresholds />
      <FanSpeedPercentageOverrides>
        <FanSpeedPercentageOverride>
          <FanSpeedPercentage>20</FanSpeedPercentage>
          <FanSpeedValue>40</FanSpeedValue>
          <TargetOperation>ReadWrite</TargetOperation>
        </FanSpeedPercentageOverride>
        <FanSpeedPercentageOverride>
          <FanSpeedPercentage>40</FanSpeedPercentage>
          <FanSpeedValue>70</FanSpeedValue>
          <TargetOperation>ReadWrite</TargetOperation>
        </FanSpeedPercentageOverride>
        <FanSpeedPercentageOverride>
          <FanSpeedPercentage>60</FanSpeedPercentage>
          <FanSpeedValue>77</FanSpeedValue>
          <TargetOperation>ReadWrite</TargetOperation>
        </FanSpeedPercentageOverride>
        <FanSpeedPercentageOverride>
          <FanSpeedPercentage>80</FanSpeedPercentage>
          <FanSpeedValue>83</FanSpeedValue>
          <TargetOperation>ReadWrite</TargetOperation>
        </FanSpeedPercentageOverride>
        <FanSpeedPercentageOverride>
          <FanSpeedPercentage>100</FanSpeedPercentage>
          <FanSpeedValue>90</FanSpeedValue>
          <TargetOperation>ReadWrite</TargetOperation>
        </FanSpeedPercentageOverride>
      </FanSpeedPercentageOverrides>
    </FanConfiguration>
  </FanConfigurations>
  <RegisterWriteConfigurations />
</FanControlConfigV2>
```

**Comment by hirschmann** [comment]
Thank you very much for sharing your config. 👍 
I can't add it to the repository, because as you said it is not very smooth.

If you manage to create a better working config in the feature, please feel free to update this issue.

**Comment by smtrejo** [comment]
I'll give it a try on a HP Pavilion DV7 1240us !!!!!

**Comment by smtrejo** [comment]
I confirm that the config "HP Compaq 6735s Turion X2 RM-72.xml" works partially. Same constant speed-up/speed-down as nickbrick's config.

**Comment by nickbrick** [comment]
@smtrejo Hey, thanks for taking the time to try my config! My dv7 (1030ev) also has a fan with only 2 wires, but I'm not sure that has anything to do with the problem, as the fan is evidently able to achieve different speeds. 

Now, I haven't looked into this since my original post, so no progress made on my part. If you want to give it a try, let me share with you my findings: In my ACPI explorer I found 2 important registers under _SB.PCI0.LPC.EC0. They are CTMP and FNSP. CTMP is the one containing the temperature (in °C) of the Thermal Zone around the fan. It is the register I manipulated to trick the fan into spinning faster or slower (through trial and error I located the thresholds where the fan jumps to its next speed). FNSP contains, I suppose, the fan speed in %. When I raise CTMP to, say, 90, I can hear my fan spin faster, and see the value of FNSP rise to 93(%). But because, as you saw yourself, the value of CTMP gets updated by the actual thermal sensor inside the laptop, it jumps between my temperature of 80 and the natural temperature in the sensor, so FNSP jumps too. Of course, if my laptop were to get hot by itself, I would observe the expected behaviour, a constant higher FNSP. Anyway, the block for me is that when I manipulate FNSP, nothing happens. Even if I set it to 93%, the fan doesn't respond, and FNSP doesn't update its value. It seems that FNSP only changes when CTMP passes one of its thresholds and the fan speed changes.
